### PR TITLE
no reason anymore to force pip at low version

### PR DESCRIPTION
**Pull request by doutriaux1**
we used to do that because of firewall forcing us to pass certificates to pip
but now we send sources to pip so the problem does not exist any longer

### CODE REVIEW

**Comment by doutriaux1** [comment]
@aashish24 @sankhesh should be an easy one to approve

**Comment by sankhesh** [comment]
Hmm.. These changes reproduce #1192.

**Comment by doutriaux1** [comment]
wow! changing pip version triggers this! This is weird...

**Comment by doutriaux1** [comment]
i guess the good news is we can close the other issue once we figured this out... :sob:

**Comment by doutriaux1** [comment]
@sankhesh what linux type is garant?

**Comment by sankhesh** [comment]
Ubuntu 14.04

**Comment by doutriaux1** [comment]
ok let me try again on my 15.04. and if it still passes i will try on a VM. Can you try on your end as well?

Thanks.

**Comment by sankhesh** [comment]
:ok:

**Comment by doutriaux1** [comment]
@sankhesh I'm looking at some of my pip log we might want to try to set: zip_safe

**Comment by doutriaux1** [comment]
my ubunutu 15.04 works like a charm (twice now) so is my mac... so is the travis linux... hum...

**Comment by doutriaux1** [comment]
the weird thing is that neither cython nor scipy seem to use pip...

**Comment by sankhesh** [comment]
What do you mean by `set zip_safe`?

Could you please push a patch with that? I can try it out on my Ubuntu 14.04.

**Comment by jbeezley** [comment]
`zip_safe` is an option in `setuptools` for packaging.  Python can run packages that are compressed into a single file in some cases.  You would want `zip_safe=False` when the package contains any compiled extension modules or data files.  In general, it is the safer choice.

**Comment by aashish24** [comment]
On travis it seems like it just got stuck. I don't see any real errors. @sankhesh do you? The SCIPY error is interesting. @jbeezley is it possible to run this branch manually on garant just to confirm?

**Comment by jbeezley** [comment]
I just triggered a rebuild on garant and the build succeeded this time.  Looks like we have a nondeterministic failure.

**Comment by aashish24** [comment]
thanks @jbeezley. @sankhesh I think the cython issue is a real issue and is un-related to this branch. I suggest we merge this branch unless you have objections to it.

**Comment by sankhesh** [comment]
Here is the error from the first build on garant: https://open.cdash.org/viewBuildError.php?buildid=3939230

```
  Configuring SCIPY:

  /home/buildbot/buildbot/uvcdat-garant-linux-release/build/build/SCIPY

  Traceback (most recent call last):

    File "<string>", line 1, in <module>

  ImportError: No module named Cython.Compiler.Main

  Traceback (most recent call last):

    File "/home/buildbot/buildbot/uvcdat-garant-linux-release/build/build/SCIPY/tools/cythonize.py", line 198, in <module>
      main()
    File "/home/buildbot/buildbot/uvcdat-garant-linux-release/build/build/SCIPY/tools/cythonize.py", line 194, in main
      find_process_files(root_dir)
    File "/home/buildbot/buildbot/uvcdat-garant-linux-release/build/build/SCIPY/tools/cythonize.py", line 186, in find_process_files
      process(cur_dir, fromfile, tofile, function, hash_db)
    File "/home/buildbot/buildbot/uvcdat-garant-linux-release/build/build/SCIPY/tools/cythonize.py", line 160, in process
      processor_function(fromfile, tofile)
    File "/home/buildbot/buildbot/uvcdat-garant-linux-release/build/build/SCIPY/tools/cythonize.py", line 80, in process_pyx
      raise Exception("Cython either isn't installed or it failed.")

  Exception: Cython either isn't installed or it failed.

  Traceback (most recent call last):

    File "setup.py", line 253, in <module>
      setup_package()
    File "setup.py", line 246, in setup_package
      generate_cython()
    File "setup.py", line 164, in generate_cython
      raise RuntimeError("Running cythonize failed!")

  RuntimeError: Running cythonize failed!

  CMake Error at
  /home/buildbot/buildbot/uvcdat-garant-linux-release/build/CMake/SCIPY_configure_step.cmake:58
  (message):

    Error in config of SCIPY
```

**Comment by jbeezley** [comment]
Could it be a race condition on the dependency resolution?  Does scipy depend on cython in the CMake files?

**Comment by sankhesh** [comment]
[Yes it does] (https://github.com/UV-CDAT/uvcdat/blob/master/CMake/cdat_modules/scipy_deps.cmake#L1)

**Comment by jbeezley** [comment]
Another possibility is that cython requires something that is missing at the time it is installed.  This could make cython use the wrong compile/link flags.

**Comment by sankhesh** [comment]
I've triggered a build on my Ubuntu 14.04 machine as well to see if I can reproduce the issue.

**Comment by aashish24** [comment]
@sankhesh seems like cython didn't build or scipy failed to find it. Can you check if Cython was built when SCIPY reported missing it?

**Comment by sankhesh** [comment]
The build was fine on my Ubuntu machine.: https://open.cdash.org/buildSummary.php?buildid=3940600

**Comment by aashish24** [comment]
Okay to merge then?

**Comment by doutriaux1** [comment]
thanks @sankhesh I'm still a bit worried about this random scipy error... It's hard to fix when we can't reproduce. It does seem like a race condition...
